### PR TITLE
Feature/45316

### PR DESCRIPTION
**Pull request by szafran89**

### CODE REVIEW

File: Magento_Company/templates/company/management/profile/edit.phtml [hunk 1]
```diff
@@ -0,0 +1,258 @@
...
+                    <?= $block->escapeHtml(__('State/Province')) ?>
+                </label>
+                <div class="control select _required">
+                    <select
+                        name="region_id"
+                        id="region_id"
+                        class="select__field select__field--native"
+                        data-validate="{ required: true }"
+                    >
+                        <option value="">
```
Comment (Aekal) [inline Magento_Company/templates/company/management/profile/edit.phtml]: Add `disabled selected` attributes

File: Magento_Company/templates/company/account/create.phtml [hunk 2]
```diff
@@ -0,0 +1,337 @@
...
+                    <?= $block->escapeHtml(__('State/Province')) ?>
+                </label>
+                <div class="control _required">
+                    <select
+                        name="company[region_id]"
+                        id="region_id"
+                        data-validate="{ required: true }"
+                        class="select__field select__field--native"
+                    >
+                        <option value="">
```
Comment (Aekal) [inline Magento_Company/templates/company/account/create.phtml]: Add `disabled selected` attributes

File: Magento_Company/templates/company/account/create.phtml [hunk 3]
```diff
@@ -0,0 +1,337 @@
...
+                    value=""
+                    title="<?= $block->escapeHtml(__('Phone Number')) ?>"
+                    class="input__field"
+                    data-validate="{ required: true }"
+                >
+            </div>
+        </fieldset>
+        <fieldset class="fieldset registration__fieldset">
+            <h3>
+                <?= /* @noEscape */ __('Company Administrator') ?>
```
Comment (Aekal) [inline Magento_Company/templates/company/account/create.phtml]: I think you can remove all occurrences of `/* @noEscape */`

File: Magento_Company/templates/company/management/dialog/customer/add.phtml [hunk 4]
```diff
@@ -0,0 +1,135 @@
+<?php
+// @codingStandardsIgnoreFile
+?>
+<?php /** @var $block Magento\Company\Block\Company\Management\Add */ ?>
+<form
+    class="form form-add-user" method="post" id="add-user-form"
```
Comment (Aekal) [inline Magento_Company/templates/company/management/dialog/customer/add.phtml]: Attributes to new lines

File: Magento_Company/templates/company/management/dialog/customer/add.phtml [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+    action="<?= $block->escapeUrl($block->getUrl('*/customer/manage')) ?>"
+    data-mage-init='{ "validation":{} }'
+>
+    <input type="hidden" name="target_id">
+    <input type="hidden" name="customer_id">
+    <fieldset class="fieldset info">
+        <div class="field input jobtitle required">
+            <label class="label" for="jobtitle">
+                <span><?= $block->escapeHtml(__('Job Title')) ?></span>
```
Comment (Aekal) [inline Magento_Company/templates/company/management/dialog/customer/add.phtml]: Remove span tags

File: Magento_Company/templates/company/management/dialog/customer/add.phtml [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+                id="jobtitle"
+                value=""
+                title="<?= $block->escapeHtml(__('Job Title')) ?>"
+                class="input-text input__field"
+                data-validate="{ required:true }"
+            >
+        </div>
+        <div class="field select role required">
+            <label class="label" for="jobtitle">
+                <span><?= $block->escapeHtml(__('User Role')) ?></span>
```
Comment (Aekal) [inline Magento_Company/templates/company/management/dialog/customer/add.phtml]: Remove span tags

File: Magento_Company/templates/company/management/dialog/customer/add.phtml [hunk 7]
```diff
@@ -0,0 +1,135 @@
...
+                id="lastname"
+                value=""
+                title="<?= $block->escapeHtml(__('Last Name')) ?>"
+                class="input-text input__field"
+                data-validate="{ required: true }"
+            >
+        </div>
+        <div class="field input email required">
+            <label class="label" for="email">
+                <span><?= $block->escapeHtml(__('Email')) ?></span>
```
Comment (Aekal) [inline Magento_Company/templates/company/management/dialog/customer/add.phtml]: Remove span tags

File: Magento_Company/templates/company/management/dialog/team/add.phtml [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+        <div class="input description">
+            <label class="label" for="description">
+                <?= $block->escapeHtml(__('Description')) ?>
+            </label>
+            <textarea
+                name="description"
+                id="description"
+                rows="4"
+                title="<?= $block->escapeHtml(__('Description')) ?>"
+                class="textarea input__field input__field--textarea"></textarea>
```
Comment (Aekal) [inline Magento_Company/templates/company/management/dialog/team/add.phtml]: Move closing `></textarea>` to new link

File: Magento_Company/templates/company/management/profile/edit.phtml [hunk 9]
```diff
@@ -0,0 +1,258 @@
...
+    $formMessages = $block->getFormMessages();
+    $countries = $block->getCountriesList();
+    $company = $block->getCustomerCompany();
+?>
+
+<?php if (!empty($formMessages)): ?>
+    <div id="company_edit_form_messages">
+        <?php foreach($formMessages as $formMessage): ?>
+            <div class="message error">
+                <span><?= $block->escapeHtml($formMessage) ?></span>
```
Comment (Aekal) [inline Magento_Company/templates/company/management/profile/edit.phtml]: Remove span tags

File: Magento_Company/templates/company/management/profile/edit.phtml [hunk 10]
```diff
@@ -0,0 +1,258 @@
...
+                <div class="control select _required">
+                    <?php $countryValue = $company->getCountryId(); ?>
+                    <select
+                        name="country_id"
+                        id="country_id"
+                        class="select__field select__field--native"
+                        title="<?= $block->escapeHtml(__('Country')) ?>"
+                        data-validate="{ required: true }"
+                    >
+                        <option value=""></option>
```
Comment (Aekal) [inline Magento_Company/templates/company/management/profile/edit.phtml]: Can we remove this empty option?

File: Magento_Company/templates/company/users/grid/buttons.phtml [hunk 11]
```diff
@@ -0,0 +1,29 @@
+<?php
+// @codingStandardsIgnoreFile
+/** @var $block Magento\Company\Block\Company\Management\Info */
+?>
+<?php if ($block->hasCustomerCompany() && $block->isUserEditAllowed()): ?>
+    <div class="dashboard-items">
```
Comment (Aekal) [inline Magento_Company/templates/company/users/grid/buttons.phtml]: Do we need all the divs below?

File: Magento_Company/web/template/users/grid/filters/filters.html [hunk 12]
```diff
@@ -0,0 +1,39 @@
...
+    <div
+        class="data-grid-filters-wrap action__handler"
+        data-part="filter-form"
+        css="_show: hasVisible()"
+    >
+        <div
+            class="field"
+            outereach="getPlain()"
+            visible="false"
+            render=""
```
Comment (Aekal) [inline Magento_Company/web/template/users/grid/filters/filters.html]: Probably this `render` can be removed

File: Magento_CompanyPayment/templates/company/management/profile/payment-method-list.phtml [hunk 13]
```diff
@@ -0,0 +1,34 @@
...
+                        <li>
+                            <?= $block->escapeHtml($paymentMethod) ?>
+                        </li>
+                    <?php endforeach ?>
+                </ul>
+            </div>
+        <?php else: ?>
+            <div class="message message-notice notice">
+                <div data-ui-id="messages-message-notice">
+                <?= $block->escapeHtml(__('This company has no payment methods. Please contact store administrator.')) ?>
```
Comment (Aekal) [inline Magento_CompanyPayment/templates/company/management/profile/payment-method-list.phtml]: Indentation

File: Magento_RequisitionList/templates/requisition/actions/rename.phtml [hunk 14]
```diff
@@ -0,0 +1,63 @@
+<div
+    class="requisition-list-title title-wrapper "
```
Comment (Aekal) [inline Magento_RequisitionList/templates/requisition/actions/rename.phtml]: Remove space after class name

File: Magento_RequisitionList/templates/requisition/view/items/grid.phtml [hunk 15]
```diff
@@ -0,0 +1,100 @@
...
+
+?>
+<div
+    class="requisition-list-view"
+    data-mage-init='{ "requisitionActions": {} }'
+>
+    <?= $block->getChildHtml('requisition.management') ?>
+    <?php if (empty($itemsList)): ?>
+        <div class="message info empty">
+            <span>
```
Comment (Aekal) [inline Magento_RequisitionList/templates/requisition/view/items/grid.phtml]: Remove span tags

File: Magento_RequisitionList/templates/requisition/view/items/grid.phtml [hunk 16]
```diff
@@ -0,0 +1,100 @@
...
+            >
+                <table class="data-table table">
+                    <thead>
+                        <tr class="headings">
+                            <th class="col number">
+                                <span>#</span>
+                            </th>
+                            <th class="col product" colspan="2">
+                                <span
+                                    data-print-label="<?= $block->escapeHtml(__('Product Name')) ?>"
```
Comment (Aekal) [inline Magento_RequisitionList/templates/requisition/view/items/grid.phtml]: No need for breaking a line

File: Magento_RequisitionList/templates/requisition/view/items/grid.phtml [hunk 17]
```diff
@@ -0,0 +1,100 @@
...
+                                >
+                                    <?= $block->escapeHtml(__('Item')) ?>
+                                </span>
+                            </th>
+                            <th class="col price">
+                                <span>
```
Comment (Aekal) [inline Magento_RequisitionList/templates/requisition/view/items/grid.phtml]: Remove span tag here and below

File: Magento_RequisitionList/templates/requisition/view/items/item.phtml [hunk 18]
```diff
@@ -0,0 +1,118 @@
...
+        <?php endif ?>
+        <div class="product-item-sku">
+            <b><?= $block->escapeHtml(__('SKU')) ?>:</b>
+            <span><?= /* @noEscape */ $item->getSku() ?></span>
+        </div>
+        <?php $block->getChildBlock('requisition.list.item.options')->setItem($item); ?>
+        <?= /* @noEscape */ $block->getChildHtml('requisition.list.item.options', false) ?>
+        <?php if ($block->getItemError()): ?>
+            <div class="message error item-error">
+                <span><?= /* @noEscape */ $block->getItemError() ?></span>
```
Comment (Aekal) [inline Magento_RequisitionList/templates/requisition/view/items/item.phtml]: Remove span tags

File: Magento_RequisitionList/templates/requisition/view/items/item.phtml [hunk 19]
```diff
@@ -0,0 +1,118 @@
...
+    </div>
+    <div class="product-item-image">
+        <img src="<?= $block->escapeUrl($block->getImageUrl()) ?>">
+    </div>
+</td>
+<td class="col price" data-th="<?= $block->escapeHtml(__('Price')) ?>">
+    <?php if (!$block->isOptionsUpdated() && $block->getRequisitionListProduct()): ?>
+        <?= /* @noEscape */ $block->getFormattedPrice() ?>
+        <?php if ($block->displayBothPrices()): ?>
+            <span class="price-excluding-tax"
```
Comment (Aekal) [inline Magento_RequisitionList/templates/requisition/view/items/item.phtml]: Move attr to new lines

File: Magento_RequisitionList/templates/requisition/view/items/item.phtml [hunk 20]
```diff
@@ -0,0 +1,118 @@
...
+        type="number"
+        size="4"
+        class="input-text qty"
+        maxlength="12"
+        data-validate='{
+            "validate-number": true,
+            "validate-greater-than-zero": true,
+            "required": true,
+            "integer": true
+            }'
```
Comment (Aekal) [inline Magento_RequisitionList/templates/requisition/view/items/item.phtml]: Indentation <-

File: Magento_RequisitionList/templates/requisition/view/items/item.phtml [hunk 21]
```diff
@@ -0,0 +1,118 @@
...
+                <button
+                    type="button"
+                    title="<?= $block->escapeHtml(__('Edit item')) ?>"
+                    data-action="edit-item"
+                    data-update-item='{
+                        "editItemUrl":"<?= $block->escapeUrl($block->getItemConfigureUrl()) ?>"
+                    }'
+                    class="action action-edit"
+                >
+                    <span><?= $block->escapeHtml(__('Edit item')) ?></span>
```
Comment (Aekal) [inline Magento_RequisitionList/templates/requisition/view/items/item.phtml]: Remove span tags

File: Magento_RequisitionList/templates/requisition/view/items/item.phtml [hunk 22]
```diff
@@ -0,0 +1,118 @@
...
+                data-action="remove-item"
+                data-delete-list='{
+                    "deleteUrl":"<?= $block->escapeUrl(
+                        $block->getUrl('*/item/delete', ['requisition_id' => $block->getRequest()->getParam('requisition_id')])
+                    ) ?>",
+                    "itemId": "<?= $block->escapeHtml($item->getId()) ?>"
+                }'
+                class="action action-delete"
+            >
+                <span><?= $block->escapeHtml(__('Remove item')) ?></span>
```
Comment (Aekal) [inline Magento_RequisitionList/templates/requisition/view/items/item.phtml]: Remove span tags

File: Magento_RequisitionList/templates/requisition/view/items/toolbar.phtml [hunk 23]
```diff
@@ -0,0 +1,24 @@
+<?php
+/**
+ * @var $block \Magento\RequisitionList\Block\Requisition\View\Details
+ */
+?>
+<div class="requisition-toolbar">
+    <div class="requisition-toolbar-select">
+        <input
+            type="checkbox"
+            name=""
```
Comment (Aekal) [inline Magento_RequisitionList/templates/requisition/view/items/toolbar.phtml]: Remove empty name

File: Magento_RequisitionList/templates/requisition/view/items/toolbar/massaction/remove.phtml [hunk 24]
```diff
@@ -0,0 +1,43 @@
...
+<button
+    type="button"
+    title="<?= $block->escapeHtml(__('Remove Selected')) ?>"
+    data-mobile-label="<?= $block->escapeHtml(__('Remove')) ?>"
+    class="action remove-selected"
+    data-bind="click: applyAction.bind($data, {
+        requisition_id: '<?= /* @noEscape */ $block->getRequest()->getParam('requisition_id') ?>'
+    })"
+>
+    <span>
```
Comment (Aekal) [inline Magento_RequisitionList/templates/requisition/view/items/toolbar/massaction/remove.phtml]: Remove span tags

File: composer.json [hunk 25]
```diff
@@ -6,7 +6,7 @@
     "require": {
       "magento/framework": "*",
       "snowdog/frontools": "^1.5",
-      "snowdog/module-alpaca-components": "dev-develop",
+      "snowdog/module-alpaca-components": "dev-feature/45316",
```
Comment (Aekal) [inline composer.json]: Don't forget to change it when merging

File: Magento_B2b/web/template/grid/listing.html [hunk 26]
```diff
@@ -0,0 +1,44 @@
...
+            role="table"
+        >
+            <caption
+                class="table__caption"
+                translate="'Table of requisitions'"
+            ></caption>
+            <thead>
+                <tr
+                    role="row"
+                    each="data: getVisible(), as: '$col'" render="getHeader()"
```
[the remark this answers is not in the document]
Comment (szafran89) [reply Magento_B2b/web/template/grid/listing.html]: done

File: Magento_Company/templates/company/management/profile/edit.phtml (hunk 1 above)
Comment (szafran89) [reply Magento_Company/templates/company/management/profile/edit.phtml]: why? this is copy/past from magento core, and there are no `selected` `disabled` options

File: Magento_Company/templates/company/account/create.phtml (hunk 2 above)
Comment (szafran89) [reply Magento_Company/templates/company/account/create.phtml]: the same as above

File: Magento_Company/templates/company/management/dialog/customer/add.phtml (hunk 6 above)
Comment (szafran89) [reply Magento_Company/templates/company/management/dialog/customer/add.phtml]: done

File: Magento_Company/templates/company/management/profile/edit.phtml (hunk 10 above)
Comment (szafran89) [reply Magento_Company/templates/company/management/profile/edit.phtml]: no

File: Magento_Company/templates/company/users/grid/buttons.phtml (hunk 11 above)
Comment (szafran89) [reply Magento_Company/templates/company/users/grid/buttons.phtml]: changed structure

File: Magento_Company/web/template/users/grid/filters/filters.html (hunk 12 above)
Comment (szafran89) [reply Magento_Company/web/template/users/grid/filters/filters.html]: I don't want to override magento attr's

File: Magento_Vault/templates/cards_list.phtml [hunk 27]
```diff
@@ -12,15 +12,15 @@
                 <thead>
                     <tr>
                         <th scope="col" class="col card-number">
-                            <?php /* @noEscape */ echo __('Card Number') ?>
+                            <?php echo __('Card Number') ?>
```
Comment (Aekal) [inline Magento_Vault/templates/cards_list.phtml]: Replace `echo` with shorthand here and below

Comment (szafran89) [reply Magento_Vault/templates/cards_list.phtml]: done

**Pull request merged**